# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1234162566 18000
#      Mon Feb 09 01:56:06 2009 -0500
# Node ID 6e764b7317bd843583be6660d19f236d2a269113
# Parent  39666c970996ecef406ac5072421ff79971c7674
test/fntests.m, scripts/test/demo.m: error on fopen failures

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2009-02-09  John W. Eaton  <jwe@octave.org>
+
+	* testfun/rundemos.m: Error if fopen fails.
+
 2009-02-08  John W. Eaton  <jwe@octave.org>
 
 	* Makefile.in (install install-strip): Don't install ls-R file.
 	(uninstall): Don't uninstall ls-R file.
 
 2009-02-06  Jaroslav Hajek  <highegg@gmail.com>
 
 	* optimization/fsolve.m: Document support for complex holomorphic
diff --git a/scripts/testfun/rundemos.m b/scripts/testfun/rundemos.m
--- a/scripts/testfun/rundemos.m
+++ b/scripts/testfun/rundemos.m
@@ -64,12 +64,16 @@ function run_all_demos (directory)
 	endif
       endif
     endif
   endfor
 endfunction
 
 function retval = has_demos (f)
   fid = fopen (f);
-  str = fscanf (fid, "%s");
-  fclose (fid);
-  retval = findstr (str, "%!demo");
+  if (f < 0)
+    error ("fopen failed: %s", f);
+  else
+    str = fscanf (fid, "%s");
+    fclose (fid);
+    retval = findstr (str, "%!demo");
+  endif
 endfunction
diff --git a/test/ChangeLog b/test/ChangeLog
--- a/test/ChangeLog
+++ b/test/ChangeLog
@@ -1,8 +1,12 @@
+2009-02-09  John W. Eaton  <jwe@octave.org>
+
+	* fntests.m (hastests): Error if fopen fails.
+
 2009-01-29  John W. Eaton  <jwe@octave.org>
 
 	* test_system.m: Use isfield instead of struct_contains.
 
 2009-01-25  Thorsten Meyer  <thorsten@hexe>
 
 	* test_struct.m: Add struct array tests.
 
diff --git a/test/fntests.m b/test/fntests.m
--- a/test/fntests.m
+++ b/test/fntests.m
@@ -66,20 +66,24 @@ function print_pass_fail (n, p)
     endif
   endif
   puts ("\n");
 endfunction
 
 ## FIXME -- should we only try match the keyword at the start of a line?
 function y = hastests (f)
   fid = fopen (f);
-  str = fread (fid, "*char")';
-  fclose (fid);
-  y = (findstr (str, "%!test") || findstr (str, "%!assert")
-       || findstr (str, "%!error") || findstr (str, "%!warning"));
+  if (fid < 0)
+    error ("fopen failed: %s", f);
+  else
+    str = fread (fid, "*char")';
+    fclose (fid);
+    y = (findstr (str, "%!test") || findstr (str, "%!assert")
+	 || findstr (str, "%!error") || findstr (str, "%!warning"));
+  endif
 endfunction
 
 function [dp, dn, dxf, dsk] = run_test_dir (fid, d);
   global files_with_tests;
   global files_with_no_tests;
   lst = dir (d);
   dp = dn = dxf = dsk = 0;
   for i = 1:length (lst)
